ble/src/main/java/no/nordicsemi/android/ble/BleManager.java

```diff
@@ -509,7 +547,15 @@ protected boolean shouldClearCacheWhenDisconnected() {
 	 */
 	@NonNull
 	public ConnectRequest connect(@NonNull final BluetoothDevice device) {
-		return connect(device, ConnectRequest.PHY_LE_1M_MASK);
+		if (mCallbacks == null) {
```

Comment:
I'm thinking, nothing prevents the user from invoking 
```java
bleManager.setGattCallbacks(null);
```

after ConnectRequest has been enqueued, which will for sure mess up things. With the new _retry_ functionality in ConnectRequest, the risk is higher. 

Would it be worth to replicate this check inside an inner _BeforeCallback_ added to _ConnectRequest_? I don't why anyone would set the callback to null, but the possibility exists.

Maybe it's overengineering, but I tried to move the _bleManager.setGattCallbacks(this)_ to _before_ and this null check failed.

Reply:
Well... setting the callbacks to *null* is shooting into owns knee, isn't it? :) Also, the setter is marked as `@NonNull`.

Reply:
👍 I didn't check the @nonnull annotation

Reply:
I was also thinking of removing the callbacks completely, and adding more callbacks to the ConnectRequest. Or at least making them not required. Right now it's messed up, as there are new and old callbacks, just to make the transition to v2 easier.